verifier/src/identity.rs

```diff
@@ -161,3 +179,120 @@ impl From<TrustedMrSignerIdentity> for TrustedIdentity {
         Self::MrSigner(mr_signer)
     }
 }
+
+// Convert two separate lists of config and sw hardening advisories into [`Advisories`]
+fn mitigated_advisories_to_advisories(
+    config_advisories: &[String],
+    sw_hardening_advisories: &[String],
+) -> Advisories {
+    let config_needed = config_advisories.is_empty().not();
+    let sw_hardening_needed = sw_hardening_advisories.is_empty().not();
+    let status = match (config_needed, sw_hardening_needed) {
+        (true, true) => AdvisoryStatus::ConfigurationAndSWHardeningNeeded,
+        (true, false) => AdvisoryStatus::ConfigurationNeeded,
+        (false, true) => AdvisoryStatus::SWHardeningNeeded,
+        (false, false) => AdvisoryStatus::UpToDate,
+    };
+
+    let advisories = config_advisories
+        .iter()
+        .chain(sw_hardening_advisories.iter());
+    Advisories::new(advisories, status)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
```

Comment:
This testing might be a bit overkill since it's a common backend, but I had the logic inverted initially so wanted to be very sure...